src/components/UserSelector.tsx

```diff
@@ -1,16 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  users: User[];
+  userSelected: User | null;
+  setUserSelected: (userSelected: User) => void;
+  getPostUser: (user: User) => void;
+  setOpenPostDetails: (openPostDetails: number) => void;
+  setWriteComment: (writeComment: boolean) => void;
+  setPosts: (posts: Post[]) => void;
+  setPost: (post: Post | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  userSelected,
+  setUserSelected,
+  getPostUser,
+  setOpenPostDetails,
+  setWriteComment,
+  setPosts,
+  setPost,
+}) => {
```

Comment:
Ensure that all functions and props have explicit type definitions to enhance type safety and prevent potential runtime errors.